File: Dasa/sorting/bubbleSort.java
package sorting;

public class bubbleSort {
    public static void BubbleSort(int arr[]){
        int n=arr.length;
        for(int i=0;i<n-1;i++){
            for(int j=0;j<n-1-i;j++){
                if(arr[j]>arr[j+1]){
                    // swap
                    int temp=arr[j];
                    arr[j]=arr[j+1];
                    arr[j+1]=temp;   // swaped
                }
            }
        }
    }
    // time complexity of bubble sort=O(n^2).
    public static void main(String[] args) {
        int arr[]=new int[]{7,8,9,6,5,2,1};

        System.out.println("Original array");
        for(int num:arr){
            System.out.print(num+" ");  // priting tghe original aray
        }
        System.out.println();

        BubbleSort(arr); // calling the func

        System.out.println("Sorted array");
        for(int num:arr){
            System.out.print(num+" ");
        }


    }
    
}

File: interact os with python/Python_problem/Sorting/bubble_sort.py
def Bubble_sort(arr):
    n=len(arr)
    
    for i in range(0,n):
        for j in range(0,n-i-1):
            if arr[j]>arr[j+1]:
                arr[j],arr[j+1]=arr[j+1],arr[j]
                
    return arr

arr=[7,2,9,4,6,5]
print(Bubble_sort(arr))            
    